x-pack/plugins/gis/public/components/map/mb/view.js

```diff
@@ -185,9 +185,22 @@ export class MBMapContainer extends React.Component {
     if (!isMapReady) {
       return;
     }
+
     removeOrphanedSourcesAndLayers(this._mbMap, layerList);
-    layerList.forEach((layer) => {
-      layer.syncLayerWithMB(this._mbMap);
+    layerList.forEach(layer => {
+      if (!layer.dataHasLoadError()) {
```

Comment:
`dataHasLoadError ` maybe this should be more generic and just be `doesLayerHaveErrors`? That way it will cover all errors, not just load errors.